tests/dataimportcron_test.go

```diff
@@ -174,23 +196,36 @@ var _ = Describe("DataImportCron", func() {
 				pvc, err := f.K8sClient.CoreV1().PersistentVolumeClaims(ns).Get(context.TODO(), currentPvc.Name, metav1.GetOptions{})
 				return err == nil && pvc.UID != currentPvc.UID
 			}, dataImportCronTimeout, pollingInterval).Should(BeTrue())
+
+			By("Wait for import completion")
+			err = utils.WaitForDataVolumePhase(f.CdiClient, ns, cdiv1.Succeeded, currentImportDv)
+			Expect(err).ToNot(HaveOccurred(), "Datavolume not in phase succeeded in time")
 		}
-		if checkGarbageCollection {
-			Eventually(func() bool {
-				dvList, err := f.CdiClient.CdiV1beta1().DataVolumes(ns).List(context.TODO(), metav1.ListOptions{})
-				Expect(err).ToNot(HaveOccurred())
-				return len(dvList.Items) == int(importsToKeep)
-			}, dataImportCronTimeout, pollingInterval).Should(BeTrue())
-		}
+		By("Check garbage collection")
+		Eventually(func() bool {
+			dvList, err := f.CdiClient.CdiV1beta1().DataVolumes(ns).List(context.TODO(), metav1.ListOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			if !garbageCollection {
```

Comment:
maybe extract the expected value here before the loop, then the Evetually will be much simpler.
```
expectedVolumes = 2
if garbageCollection {
 expectedVolumes = importsToKeep
}
``` 
 else {
 }

Reply:
then you can return dvList.Items and explicitely check for expected value in assertion
`Should(BeEqual(expectedVolumes))`

Reply:
Nice. Fixing.